core/src/main/scala/com/nvidia/spark/rapids/tool/EventLogPathProcessor.scala

```diff
@@ -253,7 +260,8 @@ object EventLogPathProcessor extends Logging {
     }.getOrElse(logsWithTimestamp)
 
     val filteredLogs = if ((filterNLogs.nonEmpty && !filterByAppCriteria(filterNLogs)) ||
-      maxEventLogSize.isDefined || minEventLogSize.isDefined) {
+      maxEventLogSize.isDefined || minEventLogSize.isDefined ||
+      fsStartTime.isDefined || fsEndTime.isDefined) {
```

Comment:
I think the filtering process has become complicated. We are now filtering at two places:
- First we filter many things inside `processAllPaths()`.
- Then again at the caller (i.e QualificationMain), we perform 2nd filtering based on some other filters.

Ideally `processAllPaths()` should return all resolved event log paths
 and then there could be `filterEventLogs()` that takes takes the different filtering arguments and applies them one by one to the full set of event logs.

This could be a later refactor since this may change the signature and affect unit tests as well.